feat: add `stamploan` to `depositCollateral`

halaprix opened the pull request:
## Ticket URL

[sc-11648]

## Description of Changes

Please list the changes introduced by this PR:
- add `stamploan` to `depositCollateral`
- default to `false` on `openPosition`

## How to Test

- run tests

## PR Definition of Done

Please ensure the following requirements have been met before marking the PR as ready for review:

- [x] All checks are passing
- [x] PR is linked to a corresponding ticket
- [x] PR title is clear and concise
- [x] Code has been self-reviewed and any fixes or improvements noted (See Code review standards in Notion)
- [x] Documentation has been updated if necessary

halaprix committed:
feat: add `stamploan` to `depositCollateral`

robercano commented on packages/ajna-contracts/contracts/ajna/ajna-actions/AjnaProxyActions.sol:
```diff
@@ -94,6 +94,12 @@ contract AjnaProxyActions is IAjnaProxyActions {
         }
     }
 
+    function _stampLoan(IERC20Pool pool, bool stamploan) internal {
```
very minor: I prefer to use more descriptive names for bool variables:
- `isStamploanEnabled` or
- `stamploanEnabled` or
- `needStamploan`

Something along those lines. Also it is nice if the function name indicates a bit more the optionality, like `stampLoanIfNeeded` so it is clear from the caller context that the code may be a no-op.

Again, personal preference, but I think it clarifies the code.

halaprix replied:
makes sense, changed it

robercano reviewed (approved):
Looks good! Just a comment on variable and functions naming

halaprix committed:
chore: update addresses

jakubswierczek committed:
v0.0.51

jakubswierczek committed:
Add and adjust stamploanEnabled in strategies and types

Added a new property 'stamploanEnabled' in deposit-borrow.ts and payback-withdraw.ts and replaced the 'stamploan' property with it in ajna-dependencies.ts. This field has been added to provide user configurability for enabling/disabling loan stamping at the time of deposit and repayment.

jakubswierczek committed:
v0.4.16
